### PR TITLE
Feature: refactor image scaling mode handling 

### LINKED ISSUE #120: [Feature Request] Optional anti aliasing
Title

### PR DESCRIPTION
**Pull request by nopeless**
Resolves #120

Let me know if you need some more features. I put WIP as I believe this needs more discussion

### CODE REVIEW

**Comment by Ruben2776** [comment]
Hi there, thanks for the PR.
I'm afraid I'm not sure what you're trying to accomplish. Setting BitmapScalingMode to NearestNeighbor results in images bigger than the screen resulotion to be visually downgraded.

**Comment by nopeless** [comment]
> Hi there, thanks for the PR. I'm afraid I'm not sure what you're trying to accomplish. Setting BitmapScalingMode to NearestNeighbor results in images bigger than the screen resulotion to be visually downgraded.

I also thought this might be a concern. personally it wasn't really noticeable, but I'll switch aliasing modes on zoom anyway. Thanks for pointing it out

I do a lot of pixel art work, and I need lightweight image viewers. Windows' default viewer doesn't have a configuration to change aliasing modes.

Before
image

After
image

File: src/PicView/Views/Windows/MainWindow.xaml [hunk 1]
```diff
@@ -471,12 +471,13 @@
                         x:Name="MainImageBorder"
                         HorizontalAlignment="Center"
                         VerticalAlignment="Center"
+                        RenderOptions.BitmapScalingMode="Fant"
```
Comment (nopeless) [inline src/PicView/Views/Windows/MainWindow.xaml]: Haven't tested the effect this has. Might remove it

**Comment by Ruben2776** [comment]
I have tried your changes myself, though I've only experienced the image looking worse using it. 
However, it can be an option in the settings window. 

If you want to edit your pull request to make the edits changeable by the user, I will accept it - or I can do the changes myself.

**Comment by nopeless** [comment]
Sure, a setting would be good enough, where the user can toggle nearest neightbor, as long as its persistent. Anything I can do to help with this?

**Comment by Ruben2776** [comment]
Can you confirm if the commit b1342155058d9194cf31be89a660e6713276e36c is what you're referring to, and possibly edit or modify it with the PR if not?

**Comment by nopeless** [comment]
I'll try to confirm it tomorrow

**Comment by nopeless** [comment]
Judging from the code, it looks like the feature was implemented properly. I am excited to see this

**Comment by nopeless** [comment]
@Ruben2776 I am getting crashes on the dev branch. I believe its just a simple mistake. Along with the fix, I also refactored it so that the logic is dealt in a more conventional manner and reduced duplicate code for future refactoring.

**Comment by nopeless** [comment]
whoops, wrong file

**Comment by Ruben2776** [comment]
Thanks. Is there anything else to be added - or otherwise I will merge?

**Comment by nopeless** [comment]
@Ruben2776 It's good to go. Afaik, the application doesn't have invalid states but please check it and test it yourself as well just in case.

**Pull request merged**